modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -75,10 +86,29 @@ object GlobalL0Service {
         }
 
       def pullGlobalSnapshots: F[Either[(Hashed[GlobalIncrementalSnapshot], GlobalSnapshotInfo), List[Hashed[GlobalIncrementalSnapshot]]]] =
-        lastGlobalSnapshotStorage.getOrdinal.flatMap {
+        lastGlobalSnapshotStorage.get.flatMap {
           _.fold {
             pullLatestSnapshot.map(_.asLeft[List[Hashed[GlobalIncrementalSnapshot]]])
-          } { lastStoredOrdinal =>
+          } { lastSnapshot =>
+            val lastStoredOrdinal = lastSnapshot.ordinal
+
+            def realign(priorityPeers: NonEmptySet[PeerId]): F[Unit] =
+              for {
+                peers <- priorityPeers.toNonEmptyList
+                  .traverse(globalL0ClusterStorage.getPeer)
+                  .map(_.toList.flatten)
+
+                majorityHash <- peers
+                  .parTraverseN(10)(p => l0GlobalSnapshotClient.getHash(lastStoredOrdinal).run(p))
+                  .map(_.flatten)
+                  .map(pickMajority[List, Hash])
+
+                isAligned = majorityHash.fold(false)(lastSnapshot.hash === _)
```

Comment:
if there is no majority it means the node is not aligned, do I understand it correctly?

Reply:
well, if it will never happen, then what's the point of using optional type for majority hash? 🤔

Reply:
That code isn't there anymore.